### PR TITLE
fix: frontend banner changes

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Frontend changes to work with new boot evaluated banner flow

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1671 #done

### CODE REVIEW

File: packages/shared/src/hooks/useBanner.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const isAvailable = useMemo(() => {
+    const lastBannerDate = Date.parse(alerts?.lastBanner);
+    const lastPostDate = Date.parse(latestBanner?.banner?.timestamp);
+
+    if (Number.isNaN(lastBannerDate) || Number.isNaN(lastPostDate)) {
+      return false;
+    }
+
+    return lastPostDate > lastBannerDate;
+  }, [alerts.lastBanner, latestBanner?.banner?.timestamp]);
```
Comment (idoshamun) [inline packages/shared/src/hooks/useBanner.ts:35]: Not sure I understand this part

Comment (rebelchris) [reply packages/shared/src/hooks/useBanner.ts:35]: Ill refactor and make it more clear 🙏

Comment (rebelchris) [reply packages/shared/src/hooks/useBanner.ts]: @idoshamun Refactored a bit, it's basically the same "extra" evaluation we use for useChangelog.
To ensure you should actually see the banner based on when you last seen it.

The main idea was I think that boot could potentially get lagged behind if you didn't particularly refreshed.

File: packages/shared/src/hooks/useBanner.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    });
+  });
+
+  const dismiss = async () => {
+    if (dismissMutation.isLoading) {
+      return;
+    }
+
+    await dismissMutation.mutateAsync();
+  };
```
Comment (capJavert) [inline packages/shared/src/hooks/useBanner.ts:51]: missing `useCallback` here

**Review by capJavert — APPROVED** [review]
Looks good 🚀

**Review by idoshamun — APPROVED** [review]

**Pull request merged**